python/ray/data/datasource/file_meta_provider.py

```diff
@@ -324,3 +327,25 @@ def prefetch_file_metadata(
             return _fetch_metadata_remotely(pieces)
         else:
             return _fetch_metadata(pieces)
+
+
+def _handle_read_s3_files_error(error: OSError, paths: Union[str, List[str]]) -> str:
+    # Specially handle AWS error when reading files, to give a clearer error message
+    # to avoid confusing users. The real issue is most likely that the AWS S3 file
+    # credentials have not been properly configured yet.
+    if "AWS Error [code 15]: No response body" in str(error):
```

Comment:
We've been seeing `AWS Error [code 100]` pop up a lot as well, which is an "UKNOWN" error: https://github.com/aws/aws-sdk-cpp/blob/f05f28b6143dfee812196923bd5417aaf47e9266/aws-cpp-sdk-s3/include/aws/s3/S3Errors.h#L47

I don't know if we've characterized those failures well enough yet, so I think we can add better error handling for such code 100 errors in a future PR.

Reply:
Sure, if we see code 100 error, we can dig more in the future.